dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -608,7 +608,11 @@ HRESULT WINAPI CDesktopFolder::GetUIObjectOf(
 
     if (IsEqualIID (riid, IID_IContextMenu))
     {
-        if (_ILIsSpecialFolder(apidl[0]))
+        if (cidl == 0)
+        {
+            hr = CDefViewBckgrndMenu_CreateInstance(this, riid, &pObj);
```

Comment:
My (not very well researched) fix for this was `hr = CreateViewObject(hwndOwner, riid, &pObj);`, which ends up calling `SHCreateDefaultContextMenu`. Are the two equivalent?

Reply:
Not really,
your path ends up with an `CDefaultContextMenu` augmented with the `Directory\\Background` class,
my approach ends up with an `CDefViewBckgrndMenu`.
Which one is better, not sure.
Maybe @yagoulas has an opinion on this?

Reply:
This is definitely not correct. Why do you need to create a menu yourself here? The idea is that CDesktopFolder aggregates functionality of the three internal IShellFolder and implements the smallest logic possible